File: engine/StarterBots/JSBot/src/models/Animal.js
export class Animal {
    constructor(id, x, y, spawnX, spawnY, score, capturedCounter, distanceCovered, isViable, heldPowerUp, activePowerUp) {
        this.id = id;
        this.x = x;
        this.y = y;
        this.spawnX = spawnX;
        this.spawnY = spawnY;
        this.score = score;
        this.capturedCounter = capturedCounter;
        this.distanceCovered = distanceCovered;
        this.isViable = isViable;
        this.heldPowerUp = heldPowerUp;
        this.activePowerUp = activePowerUp;
    }
}

File: engine/ReferenceBot/Models/Animal.cs
using ReferenceBot.Enums;

namespace ReferenceBot.Models;

public class Animal
{
    public Guid Id { get; set; }
    public int X { get; set; }
    public int Y { get; set; }
    public int SpawnX { get; set; }
    public int SpawnY { get; set; }
    public int Score { get; set; }
    public int CapturedCounter { get; set; }
    public int DistanceCovered { get; set; }
    public bool IsViable { get; set; }
    public PowerUpType? HeldPowerUp { get; set; }
    public ActivePowerUp? ActivePowerUp { get; set; }
}
